### ISSUE TITLE
Equinix Metal - Add abbility to deploy servers with Layer 2/Hybrid networking

### ISSUE THREAD

**Issue by lgn87** [kubernetes-sigs/cluster-api#9078]
### What would you like to be added (User Story)?

As a user of Cluster API it would be great to be able to lunch instances on Equinix Metal with Layer 2/Hybrid networking and to be able to specify custom set of private IPs for L2 interfaces. 

### Detailed Description

By default all servers that are created on Equinix Metal via Cluster API have Layer 3 networking and there is no option when provisioning Equinix Metal cluster to specify type of networking for instances or to create additional L2 interfaces with specific local IP addresses and VLAN.  

It would be great to implement ability to specify following when provisioning Equinix Metal cluster:
- Network type (L2/L3/Hybrid)
- Creating network interfaces with specific VLAN
- IP address range for L2 interfaces

### Label(s) to be applied

/kind feature
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

**Comment by sbueringer** [comment]
I think this issue should be opened in the repository of the Equinix Metal provider.

**Comment by killianmuldoon** [comment]
@lgn87 are you happy to close this issue and/or move it to the Equinix Metal provider repo? (I think this is the correct repo, but they've had a name change - https://github.com/kubernetes-sigs/cluster-api-provider-packet)

**Comment by sbueringer** [comment]
Yup should be the right repo (checked the provider list in our book)

/transfer cluster-api-provider-packet
